### PR TITLE
Adding basic documentation to go along with the openid-connect example

### LINKED ISSUE #1356: Add "Sign in with Slack" document section
We don't have any document pages for "Sign in with Slack" feature while having an example app in the repo: https://github.com/slackapi/node-slack-sdk/tree/main/examples/openid-connect

### List any relevant URLs

- https://slack.dev/node-slack-sdk/oauth

### Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

### PR DESCRIPTION
**Pull request by filmaj**
Fixes #1356

Very basic first pass at this... is this sufficient? What is it missing?

### CODE REVIEW

File: docs/_packages/web_api.md [hunk 1]
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
 **Note**: If you're looking for a more complete solution that handles more of the OAuth process for your app, take a
 look at the [@aoberoi/passport-slack Passport Strategy](https://github.com/aoberoi/passport-slack).
 
+---
+
+### Sign in with Slack via OpenID Connect
+
+By leveraging two APIs provided by `@slack/web-api`, [`openid.connect.token`](https://api.slack.com/methods/openid.connect.token)
```
Comment (srajiang) [inline docs/_packages/web_api.md:689]:
```suggestion
**Sign in With Slack** via OpenID Connect gives users the ability to sign into your service using their Slack profile.

The `@slack/web-api` package supports the following API methods which you can use to implement **Sign in With Slack**: 

- [`openid.connect.token`](https://api.slack.com/methods/openid.connect.token)

```

File: docs/_packages/web_api.md [hunk 2]
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
...
+and [`openid.connect.userInfo`](https://api.slack.com/methods/openid.connect.userInfo), you can provide users the
```
Comment (srajiang) [inline docs/_packages/web_api.md:690]:
```suggestion

- [`openid.connect.userInfo`](https://api.slack.com/methods/openid.connect.userInfo).

```

File: docs/_packages/web_api.md [hunk 3]
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
...
+ability to sign into your service using their Slack profile.
+
+A fully functioning [application example is available](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
```
Comment (srajiang) [inline docs/_packages/web_api.md:693]:
```suggestion
Here's a fully-functioning [sample application implementation](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
```

File: docs/_packages/web_api.md [hunk 4]
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
...
+
+Full details on how the Sign in with Slack flow works, application setup, as well as helpful resources like a Sign in With Slack
```
Comment (srajiang) [inline docs/_packages/web_api.md:695]:
```suggestion
💡 To read more about how **Sign in with Slack** works, and to access helpful resources like a Sign in With Slack
```

File: docs/_packages/web_api.md [hunk 5]
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
...
+button generator and other design assets are available on the [Authentication: Sign in with Slack](https://api.slack.com/authentication/sign-in-with-slack)
```
Comment (srajiang) [inline docs/_packages/web_api.md:696]:
```suggestion
button generator and other design assets, check out: [Authentication: Sign in with Slack](https://api.slack.com/authentication/sign-in-with-slack)
```

**Review by srajiang — CHANGES_REQUESTED** [review]
Hows this for a slightly modified wording?

**Comment by seratch** [comment]
How about having a link from the oauth document page as well? https://github.com/slackapi/node-slack-sdk/issues/1356#issuecomment-961461320

**Comment by filmaj** [comment]
@srajiang thank you for your suggestions, I have incorporated them!

@seratch I added a 'limitations' section to the oauth package docs calling out the inability to use the package to implement Sign in with Slack, and linked to the new web-api package doc on the topic.

What do you all think of the latest version of these docs?

**Review by srajiang — APPROVED** [review]

**Review by seratch — APPROVED** [review]
LGTM

**Pull request merged**